subprojects/build-option/src/main/java/org/gradle/internal/buildoption/BuildOptionFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.buildoption;
+
+import java.util.List;
+
+/**
+ * Creates a list of build options for a specific context.
+ *
+ * @since 4.2
+ */
+public interface BuildOptionFactory<T> {
```

Comment:
Can just use `Factory` instead of adding a new type.

Reply:
done